lib/ecto/query/planner.ex

```diff
@@ -446,6 +446,8 @@ defmodule Ecto.Query.Planner do
       Macro.prewalk expr, fn
         {:&, meta, [join_ix]} ->
           {:&, meta, [rewrite_ix(join_ix, ix, last_ix, source_ix, inc_ix)]}
+        expr = %Ecto.Query.Tagged{type: {type_ix, type}} ->
```

Comment:
Should we guard that `type_ix` is an integer? I am not sure that it is always an integer, especially because of types such as `{:array, :string}`.

Reply:
Yeah, we definitely should.